### PR TITLE
create filter, sort, search fields

### PR DESCRIPTION
**Pull request by Ksenaaa**

### CODE REVIEW

File: src/component/filter-sort/components/RemoveFilters.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React, { FC, useContext } from 'react'
+import DeleteSweepIcon from '@material-ui/icons/DeleteSweep';
+import { FilterSortContext } from '../../../context/FilterSortContext';
```
Comment (vell28) [inline src/component/filter-sort/components/RemoveFilters.tsx:4]: add space before

File: src/pages/basket/Basket.tsx [hunk 2]
```diff
@@ -27,6 +31,10 @@ export const Basket = () => {
         setBooks(prevBooks => prevBooks.filter(book => 
             booksLocalStore.find(localBook => book.idBook === localBook.idBook && localBook.quantity )))
     }, [booksLocalStore])
+    
+    const goBackPage = useCallback(() => 
+        navigate(-1)
```
Comment (vell28) [inline src/pages/basket/Basket.tsx:36]: if we will open this page, we won't navigate history

File: server/routes/books-list.routes.js [hunk 3]
```diff
@@ -3,34 +3,102 @@ const Book = require('../models/Book')
...
+                filteredBooks = books.filter(book => {
+                    return book.bookName.toLowerCase().includes(searchText.toLowerCase().trim()) 
+                    || book.author.toLowerCase().includes(searchText.toLowerCase().trim())
+                })
+                books = filteredBooks?.length ? filteredBooks : []
+            }
+            
+            if (!!filter) {
+                const parseFilter = filter.split(';').map(option => option.split('='))
+                parseFilter.map(option => {
```
Comment (vell28) [inline server/routes/books-list.routes.js:25]: separate thi function to smaller function and moove it to `utils` file

File: server/routes/books-list.routes.js [hunk 4]
```diff
@@ -3,34 +3,102 @@ const Book = require('../models/Book')
...
+                                } else if (!!optionItemInput.from){
+                                    return valueBook >= from && filteredBooks.push(book)
+                                } else if (!!optionItemInput.to){
+                                    return valueBook <= to && filteredBooks.push(book)
+                                }
+                            })
+                        }
+                    } else if (!!filteredBooks.length) {
+                        let fullFilter = []
+                        filteredBooks.map(book => {
```
Comment (vell28) [inline server/routes/books-list.routes.js]: forEach

File: server/routes/books-list.routes.js [hunk 5]
```diff
@@ -3,34 +3,102 @@ const Book = require('../models/Book')
...
+                                if (Object.keys(optionItemInput).length === 2) {
+                                    return valueBook >= from && valueBook <= to
+                                } else if (!!optionItemInput.from){
+                                    return valueBook >= from
+                                } else if (!!optionItemInput.to){
+                                    return valueBook <= to
+                                }
+                            })
+                        } else {
+                            return books.map(book => {
```
Comment (vell28) [inline server/routes/books-list.routes.js]: forEach

File: server/routes/books-list.routes.js [hunk 6]
```diff
@@ -3,34 +3,102 @@ const Book = require('../models/Book')
...
+                        let fullFilter = []
+                        filteredBooks.map(book => {
+                            return option[1].split(',').map(item =>
+                                item === book[option[0]].toLowerCase() && fullFilter.push(book)
+                            )
+                        })
+                        filteredBooks = fullFilter
+                    } else {
+                        return books.map(book => {
+                            return option[1].split(',').map(item =>
```
Comment (vell28) [inline server/routes/books-list.routes.js]: forEach

File: server/routes/books-list.routes.js [hunk 7]
```diff
@@ -3,34 +3,102 @@ const Book = require('../models/Book')
...
+                    } else if (!!filteredBooks.length) {
+                        let fullFilter = []
+                        filteredBooks.map(book => {
+                            return option[1].split(',').map(item =>
+                                item === book[option[0]].toLowerCase() && fullFilter.push(book)
+                            )
+                        })
+                        filteredBooks = fullFilter
+                    } else {
+                        return books.map(book => {
```
Comment (vell28) [inline server/routes/books-list.routes.js]: forEach

File: src/component/filter-sort/components/RemoveFilters.tsx (hunk 1 above)
Comment (Ksenaaa) [reply src/component/filter-sort/components/RemoveFilters.tsx:4]: fixed

File: src/pages/basket/Basket.tsx (hunk 2 above)
Comment (Ksenaaa) [reply src/pages/basket/Basket.tsx:36]: discused

File: server/routes/books-list.routes.js (hunk 4 above)
Comment (Ksenaaa) [reply server/routes/books-list.routes.js]: fixed

File: server/routes/books-list.routes.js (hunk 5 above)
Comment (Ksenaaa) [reply server/routes/books-list.routes.js]: fixed

File: server/routes/books-list.routes.js (hunk 6 above)
Comment (Ksenaaa) [reply server/routes/books-list.routes.js]: fixed

File: server/routes/books-list.routes.js (hunk 7 above)
Comment (Ksenaaa) [reply server/routes/books-list.routes.js]: fixed

File: server/routes/books-list.routes.js (hunk 3 above)
Comment (Ksenaaa) [reply server/routes/books-list.routes.js:25]: fixed

**Pull request merged**